media-libs/openimageio/openimageio-2.4.15.0.ebuild

```diff
@@ -161,9 +162,7 @@ src_configure() {
 
 	if use gui; then
 		mycmakeargs+=( -DENABLE_IV=ON -DUSE_OPENGL=ON -DUSE_QT=ON )
-		if use qt6; then
-			mycmakeargs+=( -DCMAKE_DISABLE_FIND_PACKAGE_Qt5=ON )
```

Comment:
This doesn't make sense, if this variable is not used then probably the `_Qt5` part is wrong and this should be changed to reflect the Qt5 package this is actually looking for.

Reply:
The package defaults to Qt6 and doesn't even attempt to look up Qt5. See https://bugs.gentoo.org/913746.

Reply:
Does it even build with `USE="-qt6"` then? If not then these dependencies are wrong:
```
		!qt6? (
			dev-qt/qtcore:5
			dev-qt/qtgui:5
			dev-qt/qtopengl:5
			dev-qt/qtwidgets:5
		)
		qt6? (
			dev-qt/qtbase:6[gui,widgets,opengl]
		)
```
If it only supports qt6 then we should just drop all the qt5 stuff.

Did you check that this problem is also present in the older versions? I doubt this qt problem is present there.

Reply:
The `-DCMAKE_DISABLE_FIND_PACKAGE_Qt6` disables finding Qt6 when we actually want Qt5. That's not relevant for the Qt6 path that never looks for Qt5.

Reply:
The dependencies are sane and tested for all versions for both qt5 and qt6. `2.4.13` changed `qt5` to `gui` as per https://projects.gentoo.org/qa/policy-guide/use-flags.html#pg0802.

Reply:
Right, but on the older versions USE=qt5 must look for some qt5 package. Otherwise this can't build a gui at all with qt5.

Reply:
They all do?
https://github.com/gentoo/gentoo/blob/master/media-libs/openimageio/openimageio-2.4.6.0-r1.ebuild#L85-L90
https://github.com/gentoo/gentoo/blob/master/media-libs/openimageio/openimageio-2.4.12.0.ebuild#L80-L90
https://github.com/gentoo/gentoo/blob/master/media-libs/openimageio/openimageio-2.4.13.0.ebuild#L80-L90
https://github.com/gentoo/gentoo/blob/master/media-libs/openimageio/openimageio-2.4.14.0.ebuild#L80-L90
https://github.com/gentoo/gentoo/blob/master/media-libs/openimageio/openimageio-2.4.15.0.ebuild#L80-L91
https://github.com/gentoo/gentoo/blob/master/media-libs/openimageio/openimageio-2.4.16.0.ebuild#L80-L91
https://github.com/gentoo/gentoo/blob/master/media-libs/openimageio/openimageio-2.5.4.0.ebuild#L80-L91

Reply:
My point is that if the dependency on qt5 is correct and this is actually being used, then `-DCMAKE_DISABLE_FIND_PACKAGE_Qt5=ON` cannot be an unused variable since at some point somewhere cmake must look for qt5.

Reply:
It's unused because the point where it is used is in the file used to find Qt5. That file is never executed.

Reply:
Aah right, I see, now I'm less confused. Thanks.

Reply:
The relevant code is at https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/master/src/cmake/externalpackages.cmake#L260-L274 
`Qt5` is only looked for if `Qt6` is not found, or we forcefully prohibit it from being found via `-DCMAKE_DISABLE_FIND_PACKAGE_Qt6`. So the warning only shows up for `qt5`/`gui` + `qt6`.